exporters/elasticsearch/src/es_log_exporter.cc

```diff
@@ -0,0 +1,202 @@
+    body_ = std::string(response.GetBody().begin(), response.GetBody().end());
+
+    // Set the response_received_ flag to true and notify any threads waiting on this result
+    response_received_ = true;
+    cv_.notify_all();
+  }
+
+  /**
+   * A method the user calls to block their thread until the response is received, or the timeout is
+   * exceeded.
+   */
+  bool waitForResponse(unsigned int timeoutSec = 1)
+  {
+    std::mutex mutex_;
+    std::unique_lock<std::mutex> lk(mutex_);
+    cv_.wait_for(lk, std::chrono::milliseconds(1000 * timeoutSec));
+    return response_received_;
+  }
+
+  /**
+   * Returns the body of the response
+   */
+  std::string GetResponseBody()
+  {
+    if (!response_received_)
+      return "No response";
+
+    return body_;
+  }
+
+  // Virtual method definition that isn't used
+  void OnEvent(http_client::SessionState state,
+               opentelemetry::nostd::string_view reason) noexcept override
+  {}
+
+private:
+  // Define a condition variable used for blocking
+  std::condition_variable cv_;
+
+  // Whether the response from Elasticsearch has been received
+  bool response_received_ = false;
+
+  // A string to store the response body
+  std::string body_ = "";
+};
+
+ElasticsearchLogExporter::ElasticsearchLogExporter()
+    : options_{ElasticsearchExporterOptions()},
+      session_manager_{new ext::http::client::curl::SessionManager()}
+{}
+
+ElasticsearchLogExporter::ElasticsearchLogExporter(const ElasticsearchExporterOptions &options)
+    : options_{options}, session_manager_{new ext::http::client::curl::SessionManager()}
+{}
+
+std::unique_ptr<sdklogs::Recordable> ElasticsearchLogExporter::MakeRecordable() noexcept
+{
+  return std::unique_ptr<sdklogs::Recordable>(new ElasticSearchRecordable);
+}
+
+sdklogs::ExportResult ElasticsearchLogExporter::Export(
+    const nostd::span<std::unique_ptr<sdklogs::Recordable>> &records) noexcept
+{
+  // Return failure if this exporter has been shutdown
+  if (isShutdown_)
+  {
+    if (options_.console_debug_)
+      std::cout << "Export failed, exporter is shutdown" << std::endl;
+
+    return sdklogs::ExportResult::kFailure;
+  }
+
+  // Create a json array to store all the JSON log records
+  json logs = json::array();
+
+  for (auto &record : records)
+  {
+    // Convert the log record to a JSON object, and store in json array
+    // logs.emplace_back(RecordToJSON(std::move(record)));
+  }
+
+  // Create a connection to the ElasticSearch instance
+  auto session = session_manager_->CreateSession(options_.host_, options_.port_);
+  auto request = session->CreateRequest();
+
+  // Populate the request with headers and methods
+  request->SetUri(options_.index_ + "/_bulk?pretty");
+  request->SetMethod(http_client::Method::Post);
+  request->AddHeader("Content-Type", "application/json");
+  request->SetTimeoutMs(std::chrono::milliseconds(1000 * options_.response_timeout_));
+
+  // Add the request body
+  std::string body = "";
```

Comment:
is this variable not already initialized as a private member variable?

Reply:
nope, the body member variable is for the response classes response body, and this is for the request body

Reply:
ohh i see, could they maybe be named like `reponse_body` and `request_body`? idk if you think thats necessary